src/brogue/Items.c

```diff
@@ -5675,10 +5675,10 @@ boolean hitMonsterWithProjectileWeapon(creature *thrower, creature *monst, item
 
     if (thrower == &player) {
         equippedWeapon = rogue.weapon;
-        equipItem(theItem, true, NULL);
+        equipItem(theItem, true, NULL, false);
         thrownWeaponHit = attackHit(&player, monst);
         if (equippedWeapon) {
-            equipItem(equippedWeapon, true, NULL);
+            equipItem(equippedWeapon, true, NULL, false);
```

Comment:
I didn't forsee this but unfortunately these two cases are quite problematic. We have to equip the thrown weapon in order for the slaying and hit prob calculation to use that weapon, but now this won't happen if our currently equipped weapon is cursed

Reply:
_Oh_.  I bet that's what this `force` flag (the 2nd parameter) is about.  Passing `true` certainly lets cursed gear be unequipped, and many of these "quietly recalculate things" call sites pass `true` there.

Which is not to say that my earlier refactoring didn't break it.  I'll specifically check out throwing while wielding a cursed weapon.

Reply:
..I notice that in almost all call sites, the new parameter is set to `!force`.  We might simplify by just messaging only when not forceful.

Reply:
I can confirm that while wielding a cursed weapon (spear of plenty on D16 in seed 1) throwing darts still works.  If we `printf` the name of `rogue.weapon` after the initial `equipItem` and before `attackHit`, we see it is `dart`.

However the messages arising from strength checks and cursed gear still appear from the 2nd `equipItem` when we put the cursed weapon back on.  So at least that needs to be fixed.

Reply:
I'm going to try dropping the new parameter, and make the messaging conditional on just `!force`.

Reply:
Oh right! I had forgotten about the force argument. That's fine then, but good idea with making the messaging depend on it - iirc some other functions like killCreature do a similar thing